File: src/anthropic.integration.test.ts
import * as dotenv from "dotenv";
import { describe, expect, it } from "vitest";
import { AnthropicAdapter } from "./anthropic_adapter";
import { HydraEngine } from "./engine";

dotenv.config();

const apiKey = process.env.ANTHROPIC_API_KEY;

describe("AnthropicAdapter Integration", () => {
  // Only run this test if an API key is provided
  const itIfKey = apiKey ? it : it.skip;

  itIfKey(
    "should successfully decompose a topic using the real Anthropic API",
    async () => {
      const adapter = new AnthropicAdapter(apiKey as string);

      // Pre-flight check: Verify API access with a minimal call
      try {
        await adapter.decompose("test", 1);
      } catch (e) {
        console.error("Pre-flight API check failed:", e instanceof Error ? e.message : e);
        throw e;
      }

      const engine = new HydraEngine({
        initialPrompt: "The future of sustainable energy",
        depthLimit: 1,
        branchingFactor: 2,
        adapter: adapter,
      });

      const root = await engine.run();

      expect(root.status).toBe("success");
      expect(root.children).toHaveLength(2);
      expect(root.metadata.model).toContain("claude-haiku-4-5");
      expect(root.metadata.tokens).toBeGreaterThan(0);

      console.log("Integration Test Result Topic 1:", root.children[0].topic);
      console.log("Integration Test Result Topic 2:", root.children[1].topic);
    },
    30000,
  ); // Increase timeout for real API calls
});

File: go/anthropic_integration_test.go
package hydra

import (
	"os"
	"strings"
	"testing"
)

func TestAnthropicIntegration(t *testing.T) {
	apiKey := os.Getenv("ANTHROPIC_API_KEY")
	if apiKey == "" {
		t.Skip("ANTHROPIC_API_KEY not set, skipping integration test")
	}

	adapter := NewAnthropicAdapter(apiKey, "claude-haiku-4-5-20251001")

	t.Run("Pre-flight check and decomposition", func(t *testing.T) {
		engine := NewHydraEngine(HydraConfig{
			DepthLimit:      1,
			BranchingFactor: 2,
			Adapter:         adapter,
		})

		root, err := engine.Run("The future of sustainable energy")
		if err != nil {
			t.Fatalf("Engine run failed: %v", err)
		}

		if root.Status != "success" {
			t.Errorf("Expected status success, got %s", root.Status)
		}

		if len(root.Children) != 2 {
			t.Errorf("Expected 2 children, got %d", len(root.Children))
		}

		if !strings.Contains(root.Metadata.Model, "claude-haiku-4-5") {
			t.Errorf("Expected model to contain claude-haiku-4-5, got %s", root.Metadata.Model)
		}

		if root.Metadata.Tokens <= 0 {
			t.Errorf("Expected tokens > 0, got %d", root.Metadata.Tokens)
		}
	})
}
